### PR TITLE
Add a header with constants

### PR DESCRIPTION
**Pull request by jmcarcell**
Useful for https://github.com/HEP-FCC/k4SimGeant4/pull/55 and https://github.com/key4hep/k4EDM4hep2LcioConv/pull/32 and other places so that we can set the same value in all places consistently.

More constants can be added if there are suggestions.

BEGINRELEASENOTES
- Add a constant for CellIDEncoding. Usage:

``` cpp
#include "edm4hep/Constants.h"

std::cout << edm4hep::CellIDEncoding << std::endl;
```

ENDRELEASENOTES

### CODE REVIEW

File: include/Constants.h
Comment (andresailer) [inline include/Constants.h:1]: Move to `include/edm4hep/Constants.h` also in the source tree

Comment (jmcarcell) [reply include/Constants.h:1]: Done while you were writing this comment

**Review by andresailer — APPROVED** [review]
Thanks!

File: include/edm4hep/Constants.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef EDM4HEP_CONSTANTS_H
+#define EDM4HEP_CONSTANTS_H
+
+#include <string>
```
Comment (tmadlener) [inline include/edm4hep/Constants.h:22]:
```suggestion
#include <string_view>
```

File: include/Constants.h
Comment (andresailer) [reply include/Constants.h:1]: :bird:s of a :feather:

File: include/edm4hep/Constants.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace edm4hep {
+  inline constexpr std::string_view CellIDEncoding = "CellIDEncoding";
```
Comment (tmadlener) [inline include/edm4hep/Constants.h:25]:
```suggestion
  inline static constexpr auto CellIDEncoding = "CellIDEncoding";
```
I think making this a `static const char*` makes it easier to consume for things that take a `const std::string&` as  a `string_view` will lead to

```
error: invalid initialization of reference of type 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'} from expression of type 'const std::string_view' {aka 'const std::basic_string_view<char>'
```
[compiler explorer example](https://godbolt.org/z/sWaKo5vzY)

File: include/edm4hep/Constants.h (hunk 1 above)
Comment (tmadlener) [reply include/edm4hep/Constants.h:22]: Actually if we make it `const char*` we don't need any include here.

File: include/edm4hep/Constants.h (hunk 2 above)
Comment (tmadlener) [reply include/edm4hep/Constants.h]: This will also still work if functions take a `string_view` as argument.

**Review by tmadlener — APPROVED** [review]

File: include/edm4hep/Constants.h (hunk 2 above)
Comment (jmcarcell) [reply include/edm4hep/Constants.h]: I made it `const char*`; `inline` can be dropped since it's static

**Pull request merged**